html/mycakeapp/src/Controller/AuctionController.php

```diff
@@ -86,17 +86,33 @@ public function add()
 		$biditem = $this->Biditems->newEntity();
 		// POST送信時の処理
 		if ($this->request->is('post')) {
-			// $biditemにフォームの送信内容を反映
-			$biditem = $this->Biditems->patchEntity($biditem, $this->request->getData());
-			// $biditemを保存する
-			if ($this->Biditems->save($biditem)) {
-				// 成功時のメッセージ
-				$this->Flash->success(__('保存しました。'));
-				// トップページ（index）に移動
-				return $this->redirect(['action' => 'index']);
+			// アップロードを許可する拡張子を用意
+			$extensions = array('jpg', 'jpeg', 'png', 'gif', 'JPG', 'JPEG', 'PNG', 'GIF');
+			// アップロードファイルの拡張子取得
+			$getExtension = pathinfo($_FILES['image']['name'] , PATHINFO_EXTENSION);
+			if(in_array($getExtension, $extensions, true)){
+				// $biditemにフォームの送信内容を反映
+				$biditem = $this->Biditems->patchEntity($biditem, $this->request->getData());
+				// tmp.拡張子で仮置き
+				$biditem['image_path'] = "tmp.".$getExtension;
+				// $biditemを保存する
+				if($this->Biditems->save($biditem)){
+					// biditem idに名前を変更し拡張子は保持する
+					$image_path = $biditem['id'].'.'.$getExtension;
```

Comment:
```suggestion
					$image_path = $biditem['id'] . '.' . $getExtension;
```

好みもありますが、PHPの文字列結合は「.」を使うので非常に見通しが悪くなりがちです。
適切に半角文字列など挟むと可読性が上がります。 = バグに気づきやすくなります。

また下記サンプルのように記述することもできます。
```php
$fileExtension =  'jpeg';
echo "image_file.${fileExtension}";
```